pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -505,7 +480,26 @@ func manageOpenShiftAPIServerDaemonSet_v311_00_to_latest(
 		required.Spec.Template.Spec.Containers[i].Env = append(container.Env, proxyEnvVars...)
 	}
 
-	return resourceapply.ApplyDaemonSet(client, recorder, required, resourcemerge.ExpectedDaemonSetGeneration(required, generationStatus), forceRollingUpdate)
+	// we watch some resources so that our daemonset will redeploy without explicitly and carefully ordered resource creation
+	inputHashes, err := resourcehash.MultipleObjectHashStringMapForObjectReferences(
+		kubeClient,
+		resourcehash.NewObjectRef().ForConfigMap().InNamespace(operatorclient.TargetNamespace).Named("config"),
+		resourcehash.NewObjectRef().ForConfigMap().InNamespace(operatorclient.TargetNamespace).Named("image-import-ca"),
+	)
+	if err != nil {
+		return nil, false, fmt.Errorf("invalid dependency reference: %q", err)
+	}
+	inputHashes["desired.generation"] = fmt.Sprintf("%d", operatorConfig.ObjectMeta.Generation)
+	for k, v := range inputHashes {
+		annotationKey := fmt.Sprintf("operator.openshift.io/dep-%s", k)
```

Comment:
@stlaz do you see how a future iteration simply inverts this so that the manifest itself includes the deps that should be inspected and hashes built?

Reply:
You mean include them in "v3.11.0/openshift-apiserver/ds.yaml"? Could we do it right away?

Reply:
> You mean include them in "v3.11.0/openshift-apiserver/ds.yaml"? Could we do it right away?

Let's make the big cuts first so you have something reasonable to start following in the authentication-operator.